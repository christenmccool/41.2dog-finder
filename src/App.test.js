import { render, asFragment, fireEvent } from '@testing-library/react';
import App from './App';
import { MemoryRouter } from 'react-router-dom';


test('renders without crashing', () => {
  render(<MemoryRouter><App /></MemoryRouter>);
});

test("if matches snapshot", () => {
  const {asFragment} = render(<MemoryRouter><App /></MemoryRouter>);
  expect(asFragment()).toMatchSnapshot();
})

it("dogs page renders", () => {
  const { getByAltText, debug } = render(
    <MemoryRouter initialEntries={['/dogs']}>
      <App />
    </MemoryRouter>
  );
  debug();
  expect(getByAltText("Whiskey")).toBeInTheDocument();
  expect(getByAltText("Duke")).toBeInTheDocument();
  expect(getByAltText("Perry")).toBeInTheDocument();
  expect(getByAltText("Tubby")).toBeInTheDocument();
})

it("specific dog page renders", () => {
  const { getByAltText, getByText, debug } = render(
    <MemoryRouter initialEntries={['/dogs']}>
      <App />
    </MemoryRouter>
  );

  const whiskeyLink = getByAltText("Whiskey");
  fireEvent.click(whiskeyLink);
  debug();
  expect(getByText("Whiskey loves eating popcorn.")).toBeInTheDocument();
})