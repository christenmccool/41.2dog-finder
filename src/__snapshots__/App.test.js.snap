// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`if matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="Nav"
    >
      <a
        aria-current="page"
        class="active"
        href="/dogs"
      >
        Home
      </a>
      <a
        href="/dogs/Whiskey"
      >
        Whiskey
      </a>
      <a
        href="/dogs/Duke"
      >
        Duke
      </a>
      <a
        href="/dogs/Perry"
      >
        Perry
      </a>
      <a
        href="/dogs/Tubby"
      >
        Tubby
      </a>
    </div>
    <div
      class="DogList"
    >
      <a
        href="/dogs/Whiskey"
      >
        <img
          alt="Whiskey"
          src="whiskey.jpg"
        />
      </a>
      <a
        href="/dogs/Duke"
      >
        <img
          alt="Duke"
          src="duke.jpg"
        />
      </a>
      <a
        href="/dogs/Perry"
      >
        <img
          alt="Perry"
          src="perry.jpg"
        />
      </a>
      <a
        href="/dogs/Tubby"
      >
        <img
          alt="Tubby"
          src="tubby.jpg"
        />
      </a>
    </div>
  </div>
</DocumentFragment>
`;
